### PR TITLE
Ensure that `pointerUp` Action is Being Issued on Element Being Tested

### LINKED ISSUE #606: Retroactive review of user-action-pseudo-classes-in-has.html changes/regressions
### Test List

https://wpt.fyi/results/css/selectors/invalidation/user-action-pseudo-classes-in-has.html

### Rationale

https://github.com/web-platform-tests/wpt/pull/43122 rewrote this test, making it pass in Firefox, but fail in Chrome.

Here are the before/after commits:
https://wpt.fyi/results/css/selectors/invalidation/user-action-pseudo-classes-in-has.html?sha=17dcc288d1c558801914f40f07e9735adecea59f
https://wpt.fyi/results/css/selectors/invalidation/user-action-pseudo-classes-in-has.html?sha=5fab76582c597542ff5dfc02b3462531842349d8

There are no Safari runs for these exact commits, but these runs sandwich the change, and it looks like it was failing before and now in a different way:
https://wpt.fyi/results/css/selectors/invalidation/user-action-pseudo-classes-in-has.html?run_id=6308805735874560&run_id=5148500288864256

### PR DESCRIPTION
**Pull request by dshin-moz**
There does not seem to be any guarantee that the pointer will move between actions.

Fixes web-platform-tests/interop#606 - verified on local Chromium build.

### CODE REVIEW

**Comment by foolip** [comment]
@dshin-moz it looks like this breaks the test in Firefox instead:
https://wpt.fyi/results/css/selectors/invalidation/user-action-pseudo-classes-in-has.html?label=pr_head&max-count=1&pr=43214

Did you intend for that to happen?

**Comment by dshin-moz** [comment]
Hm.. That's unfortunate - Looking at the logs, the failures happen before the changed line. Checking locally  with `--verify-repeat-restart=500` and it happens before this change as well, it's just that the failure rate is 3/500 :/

File: css/selectors/invalidation/user-action-pseudo-classes-in-has.html [hunk 1]
```diff
@@ -37,47 +38,56 @@
...
-      .Actions()
-      .pointerUp()
-      .send();
-    assert_equals(getComputedStyle(subject1).color, "rgb(0, 0, 255)",
-                  "subject1 should be blue again");
-    assert_equals(getComputedStyle(subject3).color, "rgb(0, 0, 128)",
-                  "subject3 should be navy again");
+        .Actions()
+        .pointerUp()
+        .pointerMove(0, 0, {origin: unhoverme})
```
Comment (foolip) [inline css/selectors/invalidation/user-action-pseudo-classes-in-has.html:78]: This removes coverage for what was previously tested, which was to just release the mouse button without moving it. Can you keep this part of the test?

```suggestion
        .pointerMove(0, 0, {origin: hoverme})
```

And keep the assertions as they were. If you think it's also useful to test moving the mouse right after mouseup, additional steps for that would be OK, but I'm not sure I see the difference to the first unhoverme bits.

Comment (dshin-moz) [reply css/selectors/invalidation/user-action-pseudo-classes-in-has.html:78]: It does remove that coverage - but does the suggested change really test "going from `:hover:active` to `:hover`?"
I could be misreading this, but the current & suggested action sequence decomposes into two ticks, right? So it'd be a succession of "`:has(:hover:active)` invalidates" (Which is similar to the current change), then "`:has(:hover)` invalidates" (Which is already tested)?

Comment (dshin-moz) [inline css/selectors/invalidation/user-action-pseudo-classes-in-has.html:78]: Could do:
```suggestion
        .Actions()
        .pointerUp()
        .pointerMove(0, 0, {origin: unhoverme})

    // Independent action sequence that should have the last invalidation to be `:has(:hover:active)` -> `:has(:hover)`
    await new test_driver
        .Actions()
        .pointerMove(0, 0, {origin: hoverme})
        .pointerDown()
        .pointerUp()
        .send();
    // ...
```

**Review by foolip — APPROVED** [review]

File: css/selectors/invalidation/user-action-pseudo-classes-in-has.html (hunk 1 above)
Comment (foolip) [reply css/selectors/invalidation/user-action-pseudo-classes-in-has.html:80]: Now that I understand the motivation for the change, I don't think it's important to keep testing the old thing. I think we should treat `:active` and `:hover` as separate, and both are now testing in both directions (matching to not matching and vice versa)

Comment (foolip) [reply css/selectors/invalidation/user-action-pseudo-classes-in-has.html:80]: I think testing `:has(:hover:active)` &rarr; `:has(:hover)` is perhaps not needed, but you already wrote the test, so let's have it! (I deleted a previous comment saying we don't need this.)

**Pull request merged**